api/src/main/java/org/apache/iceberg/PartitionField.java

```diff
@@ -28,11 +28,13 @@
  */
 public class PartitionField implements Serializable {
   private final int sourceId;
+  private final int partitionFieldId;
   private final String name;
   private final Transform<?, ?> transform;
 
-  PartitionField(int sourceId, String name, Transform<?, ?> transform) {
+  PartitionField(int sourceId, int partitionFieldId, String name, Transform<?, ?> transform) {
```

Comment:
The name `partitionFieldId` is redundant because the class is `PartitionField`.

Let's use the same convention that is used in types. The PartitionField should have an `id` instance variable that is accessed by a `fieldId` method.

Reply:
sure.